File: Project1_Code/Main.py
import json

filename = 'resource/posts.json' # the path of json file
with open(filename) as f:
    posts = json.load(f)

for post in posts:
    print(type(post)) # it is a dist type
    print(post['postID'])
    print(json.dumps(post, indent=1))

File: Project1_Code/src/Main.java
import com.alibaba.fastjson.JSON;

import java.io.IOException;
import java.nio.file.Files;
import java.nio.file.Path;
import java.util.List;

public class Main {
    public static void main(String[] args) {
        try {
            String jsonStrings = Files.readString(Path.of("resource/posts.json"));
            List<Post> posts = JSON.parseArray(jsonStrings,Post.class);
            posts.forEach(System.out::println);

        } catch (IOException e) {
            throw new RuntimeException(e);
        }
    }
}
